### PR TITLE
feat: Server side logs 

### PR DESCRIPTION
**Pull request by adityathebe**
Addresses https://github.com/flanksource/apm-hub/issues/30

### CODE REVIEW

**Comment by moshloop** [comment]
@adityathebe can you add an integration test for this ?

**Comment by adityathebe** [comment]
@moshloop I've added integration test.

File: pkg/files/search.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		file, err := os.Open(path)
+		if err != nil {
+			return nil, fmt.Errorf("error opening file. path=%s; %w", path, err)
+		}
+
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			fileContents[path] = append(fileContents[path], logs.Result{
+				Time: fInfo.ModTime().Format(time.RFC3339),
+				// Labels: , all the records will have the same labels. Is it necessary to add it here?
```
Comment (moshloop) [inline pkg/files/search.go:59]: Yes, we should also add the filename

File: pkg/files/search.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+type logsPerFile map[string][]logs.Result
+
+// readFilesLines will take a list of file paths
+// and then return each lines of those files.
+func readFilesLines(paths []string) (logsPerFile, error) {
+	fileContents := make(logsPerFile, len(paths))
+	for _, path := range paths {
+		fInfo, err := os.Stat(path)
+		if err != nil {
+			return nil, fmt.Errorf("error get file stat. path=%s; %w", path, err)
```
[the remark this answers is not in the document]
Comment (adityathebe) [reply pkg/files/search.go]: Modified this. On error, it'll continue reading other files

**Comment by adityathebe** [comment]
@moshloop should be good now

File: pkg/files/search.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+type logsPerFile map[string][]logs.Result
+
+// readFilesLines takes a list of file paths and returns each lines of those files.
+// If labels are also passed, it'll attach those labels to each lines of those files.
+func readFilesLines(paths []string, labelsToAttach map[string]string) logsPerFile {
+	fileContents := make(logsPerFile, len(paths))
+	for _, path := range paths {
+		fInfo, err := os.Stat(path)
```
Comment (moshloop) [inline pkg/files/search.go:41]: Can we add support for globs ?

File: pkg/files/search.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			logger.Warnf("error opening file. path=%s; %w", path, err)
+			continue
+		}
+
+		// All lines of the same file will share these labels
+		labels := mergeMap(map[string]string{"filepath": path}, labelsToAttach)
```
Comment (moshloop) [inline pkg/files/search.go:54]:
```suggestion
		labels := mergeMap(map[string]string{"path": path}, labelsToAttach)
```

File: pkg/files/search.go (hunk 3 above)
Comment (adityathebe) [reply pkg/files/search.go:42]: Added support

File: pkg/files/search.go [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+			return false
+		}
+	}
+
+	return true
+}
+
+// MergeMap will merge map b into a.
+// On key collision, map b takes precedence.
+func MergeMap(a, b map[string]string) map[string]string {
```
Comment (adityathebe) [inline pkg/files/search.go:82]: @moshloop Do we have a Go module with small helper funcs like this one? Don't want this in here. I could create pkg/helper and move it there.

Comment (adityathebe) [reply pkg/files/search.go:82]: I think it could go in here e- https://github.com/flanksource/commons/blob/master/collections/collections.go

**Pull request merged**